app/DAO/BakeryDB.scala

```diff
@@ -62,6 +62,35 @@ class BakeryDB {
       .unsafeRunSync()
   }
 
+  def editByID(
+      id: UUID,
+      name: Option[String],
+      quantity: Option[Int],
+      price: Option[Double]
+  ): Option[Product] = {
+    val nameFragment = name.getOrElse(None) match {
+      case None       => ""
+      case Some(name) => " ,".concat(name.toString)
+    }
+    val quantityFragment = quantity.getOrElse(None) match {
+      case None           => ""
+      case Some(quantity) => " ,".concat(quantity.toString)
+    }
+    val priceFragment = price.getOrElse(None) match {
+      case None        => ""
+      case Some(price) => " ,".concat(price.toString)
+    }
```

Comment:
It's generally poor practice to build string on your own, especially when you are using it to access a database. For example, if I sent an update json with the form:
```JSON
{
 "name":  "'randomProduct'; DROP TABLE Product --"
}
```
And you wouldn't be having any product database table anymore.  Try to use the interpolated value support (`?` syntax in the database) or the `set = ` syntax in the query or even the (value1, value2) syntax (example from the doobie website: `insert into person (name, age) values ($name, $age)`). In general it would be best to handle each specific instance of the match statement in a single query producing result so you can run that dynamically produced query

Reply:
Ok thank you for the explanation, I felt like it should be done another way but didn't know how to go about it